File: taurus-fe/src/app/service/albums.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { RoleEnums } from '../constants';
import { Albums, AlbumsCriteria } from '../module';
import { CommonOpenSearchService } from './common-open-search.service';
import { KeycloakService } from './keycloak.service';

@Injectable({
    providedIn: 'root'
})
export class AlbumsService extends CommonOpenSearchService<Albums, AlbumsCriteria> {

    constructor(protected override readonly http: HttpClient, private readonly keycloakService: KeycloakService) {
        super(http);
    }

    override resourceName(): string {
        if (this.keycloakService.currentUserRoles.every(role => role === RoleEnums.USER)) {
            return "user/albums";
        }

        return "albums";
    }
}

File: taurus-be/src/main/java/com/fundaro/zodiac/taurus/service/user/AlbumsService.java
package com.fundaro.zodiac.taurus.service.user;

import com.fundaro.zodiac.taurus.domain.Albums;
import com.fundaro.zodiac.taurus.domain.criteria.AlbumsCriteria;
import com.fundaro.zodiac.taurus.service.dto.AlbumsDTO;

public interface AlbumsService extends CommonOpenSearchService<Albums, AlbumsDTO, AlbumsCriteria> {
}
